cpp/src/arrow/compute/exec/asof_join_benchmark.cc

```diff
@@ -49,6 +49,20 @@ static Result<TableStats> MakeTable(const TableGenerationProperties& properties)
   return Result<TableStats>({table, rows, rows * row_size});
 }
 
+// As opposed to using table_source, we create a make a ReaderGenerator for the specified
+// table. This allows us to specify a thread pool to isolate the threads used for each
+// source as an anti-deadlocking mechanism.
+static ExecNode* MakeTableSourceNode(std::shared_ptr<arrow::compute::ExecPlan> plan,
+                                     std::shared_ptr<Table> table, int batch_size,
+                                     arrow::internal::ThreadPool* thread_pool) {
+  std::shared_ptr<TableBatchReader> reader = std::make_shared<TableBatchReader>(table);
```

Comment:
```suggestion
  std::shared_ptr<TableBatchReader> reader = std::make_shared<TableBatchReader>(std::move(table));
```
Another nitpick to avoid a shared_ptr copy.